File: tutorials/tour-of-restate-typescript/src/app/checkout_service.ts
/*
 * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Restate examples,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/
 */

import * as restate from "@restatedev/restate-sdk";

export const checkoutService = restate.service({
  name: "CheckoutService",
  handlers: {
    async handle(ctx: restate.Context, request: { userId: string; tickets: string[] }){
      console.info("Hello")
      return true;
    },
  }
});

export const CheckoutService: typeof checkoutService = { name: "CheckoutService"};

File: go/tutorials/tour-of-restate-go/part2/checkoutservice.go
package main

import restate "github.com/restatedev/sdk-go"

type CheckoutService struct{}

type CheckoutRequest struct {
	UserId  string   `json:"userId"`
	Tickets []string `json:"tickets"`
}

func (CheckoutService) Handle(ctx restate.Context, request CheckoutRequest) (bool, error) {
	return true, nil
}
